boot.js

```diff
@@ -350,8 +350,11 @@ function closeWithCbOrNextTick (func, cb, context) {
 function encapsulateTwoParam (func, that) {
   return _encapsulateTwoParam.bind(that)
   function _encapsulateTwoParam (context, cb) {
-    if (func.length === 0 || func.length === 1) {
-      func(this)
+    if (func.length === 0) {
+      func()
+      process.nextTick(cb)
+    } else if (func.length === 1) {
+      func(cb)
```

Comment:
You are changing the behavior here. Can you restore `this`?

Reply:
I did that because of the original issue in fastify stated 

>if onClose(func) is called with a function with 1 parameter, pass the callback and not the >instance as the first one. 

Restored now